### PR TITLE
Add Flatten with Duplicate Inputs Test

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This test demonstrates that if a Flatten.PCollections transform has the
same PCollectionList in its input multiple times, it will duplicate the
input PCollection for each time it occurs within the list.

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java [hunk 1]
```diff
@@ -114,6 +115,25 @@ public void testFlattenPCollectionListEmpty() {
     PAssert.that(output).empty();
     p.run();
   }
+  @Test
```
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java]: Will add new line before `@Test` before merging

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java [hunk 2]
```diff
@@ -114,6 +115,25 @@ public void testFlattenPCollectionListEmpty() {
...
+        .and(lines2)
+        .and(lines)
+        .and(lines)
+        .apply(Flatten.<String>pCollections());
+
+    PAssert.that(flattened).containsInAnyOrder(Iterables.concat(LINES, LINES, LINES, LINES2));
+
+    p.run();
+  }
+
```
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java]: Will remove this additional line

**Comment by lukecwik** [comment]
LGTM, will merge with minor changes

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java]: Done.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java (hunk 2 above)
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java]: Done.

**Pull request closed without merging**